File: tmp/list_tables.py
import os
import psycopg2
from dotenv import load_dotenv

load_dotenv('.env.local')

def list_tables():
    conn_str = os.getenv("SUPABASE_CONNECTION_STRING")
    try:
        conn = psycopg2.connect(conn_str)
        with conn.cursor() as cur:
            cur.execute("SELECT table_name FROM information_schema.tables WHERE table_schema = 'public'")
            tables = cur.fetchall()
            print("Tables in 'public' schema:")
            for table in tables:
                print(f"- {table[0]}")
        conn.close()
    except Exception as e:
        print(f"Error checking tables: {e}")

if __name__ == "__main__":
    list_tables()

File: tmp/list_tables.js

import postgres from 'postgres';

const sql = postgres('postgresql://postgres.boyleodbpsgqtjbstcua:Mom11343Supapass@aws-0-us-east-1.pooler.supabase.com:6543/postgres');

async function listTables() {
  try {
    const tables = await sql`
      SELECT table_name 
      FROM information_schema.tables 
      WHERE table_schema = 'public'
      ORDER BY table_name
    `;
    console.log('--- TABLES ---');
    tables.forEach(t => console.log(t.table_name));
    console.log('--- END TABLES ---');
  } catch (err) {
    console.error('Error:', err);
  } finally {
    await sql.end();
  }
}

listTables();
